spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRToBlas.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.utils.intermediate
+
+import com.intel.analytics.bigdl._
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity, TensorModule}
+import com.intel.analytics.bigdl.optim.DistriOptimizer._
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{Node, T}
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+
+abstract class ConvertBase[T, D] {
+  /**
+    * clone node relations
+    * @param oldToNew node element maps from T to D
+    */
+  def cloneNode(allNodes: Array[Node[T]], oldToNew: mutable.HashMap[Node[T], Node[D]]): Unit = {
+    allNodes.foreach(node => {
+      node.nextNodesAndEdges.foreach(nextNodeAndEdge => {
+        if (oldToNew.contains(nextNodeAndEdge._1)) {
+          oldToNew.get(node).get.add(oldToNew.get(nextNodeAndEdge._1).get, nextNodeAndEdge._2)
+        }
+      })
+    })
+  }
+
+  def enableConvertLayer(layer: T) : Boolean
+
+  def convertLayer(layer : T) : D
+
+  def enableConvert(allNodes: Array[Node[T]]) : Boolean = {
```

Comment:
`enableConvert` seems more like we should enable the converting, here we just check , maybe `convertingEnabled` a better one ?

Reply:
yeah, maybe we should not use "enable". What about convertingCheck?

Reply:
yes, that would also work.

Reply:
ok, changed